### PR TITLE
feat(accounts-controller): add getNextAvailableAccountName public method

### PR DESCRIPTION
**Pull request by ccharly**
## Explanation

Make this function available since the logic might be tricky when we have gaps in the account list.

This can then be used by the UI if it needs to get the next account name available during account creating (from within the extension).

### `@metamask/accounts-controller`

- **CHANGED**: Make getNextAvailableAccountName method public

## Checklist

- [x] I've updated the test suite for new or updated code as appropriate
- [x] I've updated documentation (JSDoc, Markdown, etc.) for new or updated code as appropriate
- [x] I've highlighted breaking changes using the "BREAKING" category above as appropriate

### CODE REVIEW

File: packages/accounts-controller/src/AccountsController.test.ts [hunk 1]
```diff
@@ -2115,6 +2116,63 @@ describe('AccountsController', () => {
         );
         expect(account).toStrictEqual(mockAccount);
       });
+
+      describe('getNextAvailableAccountName', () => {
+        it('get the next account name', async () => {
```
Comment (gantunesr) [inline packages/accounts-controller/src/AccountsController.test.ts:2121]:
```suggestion
        it('gets the next account name', async () => {
```

File: packages/accounts-controller/src/AccountsController.test.ts [hunk 2]
```diff
@@ -2115,6 +2116,63 @@ describe('AccountsController', () => {
...
+            messenger,
+          });
+
+          const accountName = messenger.call(
+            'AccountsController:getNextAvailableAccountName',
+          );
+          expect(accountName).toBe('Account 2');
+        });
+
+        it('get the next account name with a gap', async () => {
```
Comment (gantunesr) [inline packages/accounts-controller/src/AccountsController.test.ts:2144]:
```suggestion
        it('gets the next account name with a gap', async () => {
```

**Review by gantunesr — APPROVED** [review]
LGTM. Just two minor typos

**Comment by ccharly** [comment]
@metamaskbot publish-preview

**Pull request merged**